README.md

```diff
@@ -1,5 +1,88 @@
-## csi
-CSI implementation for openebs storage engines
+### csi
+CSI driver implementation for openebs storage engines.
 
+### Overview
+OpenEBS CSI driver implementation comprises of 2 components:
+1) Controller service: Runs as stateful set
+2) Node service: Runs as a daemonset
+
+### Provision a volume using OpenEBS CSI driver
+
+Apply OpenEBS CSI Operator:
+```
+kubectl apply -f https://raw.githubusercontent.com/openebs/csi/master/deploy/csi-operator.yaml
+```
+Create a Storage Class pointing to OpenEBS CSI provisioner:
+```
+kubectl apply -f https://raw.githubusercontent.com/openebs/csi/master/deploy/sc.yaml
+```
+Create PVC with above Storage Class:
+```
+kubectl apply -f https://raw.githubusercontent.com/openebs/csi/master/deploy/pvc.yaml
+```
+Since the provisioner specified in the SC is openebs-csi.openebs.io,
+CSI Controller service recieves a Volume creation request CreateVolume() via grpc. 
+which in turn creates a CStorVolumeClaim(CVC) CR. This will be created with empty nodeID and status marked as pending. 
+And a succees response is sent immediately after creating CVC. Once this API responds success kubernetes creates a PV object and binds it to PVC
+
+The watcher for CVC CR in m-apiserver waits for node-id to be filled to provision the volume.
+```
+# Sample CVC CR on receiving VolumeCreate Request
+apiVersion: openebs.io/v1alpha1
+kind: CStorVolumeClaim
+metadata:
+  annotations:
+    openebs.io/config-class: openebs-csi-default
+    openebs.io/volumeID: pvc-b3e44fa5-98f2-11e9-b58e-42010a80006c
+  finalizers:
+  - cvc.openebs.io/finalizer
+  name: pvc-b3e44fa5-98f2-11e9-b58e-42010a80006c
+  namespace: openebs
+spec:
+  capacity:
+    storage: 40Gi
+status:
+  phase: Pending
+```
+
+Deploy a percona app with the above PVC:
+```
+kubectl apply -f https://raw.githubusercontent.com/openebs/csi/master/deploy/percona.yaml
+kubectl apply -f https://raw.githubusercontent.com/openebs/csi/master/deploy/sqltest_configmap.yaml
+```
+
+On deploying the app CSI Node Service receives a NodePublishVolume() request via grpc,
+which in turn patches nodeID to the previously created CVC CR and waits for the  status to be updated to bound by CVC watcher. 
+The bound status implies that the following required volume components have been created by CVC watcher:
+1) Target service
+2) Target deployment
+3) CstorVolume CR
+4) CstoVolumeReplica CR
+```
+# Sample CVC CR after NodePublish is successful
+apiVersion: openebs.io/v1alpha1
+kind: CStorVolumeClaim
+metadata:
+  annotations:
+    openebs.io/config-class: openebs-csi-default
```

Comment:
let us have this as `openebs.io/cstor-volume-config-class`

Reply:
Since we are not ready with cstor-volume-config-class, I think it would not be a good idea to include this in README.md. Would it be fine if we place it over here once it is implemented.